File: java/src/main/java/deltix/luminary/Int32Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class Int32Def implements TypeDef<TypeInt32>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private Int32Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeInt32.INSTANCE, new LiteralInt32(Integer.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeInt32.INSTANCE, new LiteralInt32(Integer.MAX_VALUE)));
    }

    public static final Int32Def INSTANCE = new Int32Def();

    @Override
    public String getName() {
        return "UInt32";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeInt32 getType() {
        return TypeInt32.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}

File: csharp/main/Int32Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class Int32Def : IConstantScope, ITypeDef<TypeInt32>
	{
		private Int32Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeInt32.Instance, new LiteralInt32(Int32.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeInt32.Instance, new LiteralInt32(Int32.MaxValue)));
		}

		public static Int32Def Instance = new Int32Def();

		public String Name => "Int32";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeInt32 Type => TypeInt32.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}
